### PR TITLE
Remove upper bounds on cuda-python to allow 12.6.2 and 11.8.5

### PR DESCRIPTION
**Pull request by bdice**
Now that some upstream bugs have been fixed, we can allow cuda-python 12.6.2 and 11.8.5.

See https://github.com/NVIDIA/cuda-python/issues/226#issuecomment-2472355738 for more information.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -143,8 +143,9 @@ dependencies:
         packages:
           - c-compiler
           - cxx-compiler
-          - nccl>=2.19
           - libucxx==0.42.*,>=0.0.0a0
+          - nccl>=2.19
+          - spdlog>=1.14.1,<1.15
```
Comment (bdice) [inline dependencies.yaml:148]: `spdlog` was missing in RAFT conda devcontainers. Previously it was gotten transitively from RMM but that was removed in https://github.com/rapidsai/rmm/pull/1739, causing a build failure. I discussed with @vyasr and we agreed this fix (adding `spdlog`) is necessary for the short term. In the medium-to-long term, @vyasr is going to rework the loggers.

Comment (jameslamb) [reply dependencies.yaml:148]: Thanks for summarizing that here, works for me.

(and thanks for alphabetizing this list while you're touching it).

**Review by jameslamb — APPROVED** [review]
Just adding a `spdlog` dependency for now sounds good to me, thanks.

**Review by cjnolet — APPROVED** [review]

**Pull request merged**